### PR TITLE
Swift underfs fixes

### PR DESCRIPTION
**Pull request by gilv**
This push request contains:

1.  Swift driver sent wrong authentication header when used with swiftauth. The wrong header was of the form user:tenant, while it suppose to be tenant:user. I verified that the fix is working now correctly with SoftLater object store.
2. swift.apikey is a bad naming. Swift driver should not use apikey, but to use swift.password. I added password  key and also left the old one, so the change will be backward compatible. At some point we can remove swift.apikey at all.
3. There is no need to have special configuration key for swift. The documentation clearly states how to modify alluxio-env.sh, so there is no need to have specific swift configuration.

### CODE REVIEW

File: conf/alluxio-swift-env.sh.template [hunk 1]
```diff
@@ -1,120 +0,0 @@
-#!/usr/bin/env bash
```
Comment (calvinjia) [inline conf/alluxio-swift-env.sh.template]: I think having a template is still useful despite having documentation.

**Comment by gilv** [comment]
@calvinjia But you don't have template for S3. I thought to make things more consistent. To have templates for both or remove them completely. I saw that S3 doesn't has a template. 
What do you prefer?

File: core/client/pom.xml [hunk 2]
```diff
@@ -79,6 +79,11 @@
       <artifactId>alluxio-underfs-s3</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-underfs-swift</artifactId>
+      <version>${project.version}</version>
+    </dependency>
```
Comment (calvinjia) [inline core/client/pom.xml]: We originally left this out because of the dependency requirement on the hadoop version. Has that issue been resolved?

**Comment by calvinjia** [comment]
@gilv I would prefer having a ticket to add a S3 template instead of removing the Swift template.

File: underfs/swift/pom.xml [hunk 3]
```diff
@@ -39,36 +39,6 @@
       <version>${project.version}</version>
     </dependency>
     <dependency>
-      <groupId>org.apache.hadoop</groupId>
```
Comment (calvinjia) [inline underfs/swift/pom.xml]: Is this no longer required because we use JOSS now?

File: underfs/swift/pom.xml [hunk 4]
```diff
@@ -81,7 +51,13 @@
     <dependency>
       <groupId>org.javaswift</groupId>
       <artifactId>joss</artifactId>
-      <version>0.9.8</version>
+      <version>0.9.10</version>
+      <exclusions>
+          <exclusion>
+              <groupId>org.slf4j</groupId>
+              <artifactId>slf4j-api</artifactId>
+          </exclusion>
```
Comment (calvinjia) [inline underfs/swift/pom.xml]: Is this exclusion necessary?

File: underfs/swift/pom.xml (hunk 3 above)
Comment (gilv) [inline underfs/swift/pom.xml]: @calvinjia yes, we don't need this Hadoop stuff in the swift module. We use JOSS so all those Hadoop dependencies are not needed in the Swift module.

File: underfs/swift/pom.xml (hunk 4 above)
Comment (gilv) [inline underfs/swift/pom.xml]: @calvinjia Without this exclusion i saw collisions in Spark. Seems collide with some other package. Not sure it's present in all versions of Spark. But in my case, there were warnings and exceptions from Spark, complaining on collision with this package. Exclusion resolved it.

File: core/client/pom.xml (hunk 2 above)
Comment (gilv) [inline core/client/pom.xml]: @calvinjia yes, if i understand you correctly - then Swift module doesn't need any of Hadoop dependencies anymore. Without swift lines in the client - Spark can't work with Swift.

Comment (calvinjia) [inline core/client/pom.xml]: Makes sense, thanks.

File: underfs/swift/pom.xml (hunk 4 above)
Comment (calvinjia) [inline underfs/swift/pom.xml]: The reason I ask is because we explicitly include `slf4j-api` as a top level dependency, so the transitive dependency should not be pulled in. Could you verify if the warnings occur without the exclusion but with the above inclusion?

File: underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java [hunk 5]
```diff
@@ -175,6 +187,9 @@ public boolean delete(String path, boolean recursive) throws IOException {
   @Override
   public boolean exists(String path) throws IOException {
     String newPath = stripPrefixIfPresent(path);
+    if (newPath.endsWith("_temporary")) {
+      return true;
+    }
```
Comment (calvinjia) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: There are already several fixes going into this PR, could we extract this one into another PR?

**Comment by calvinjia** [comment]
@gilv 
Several of the changes in this PR would be really good to get merged in ASAP while others need additional discussion, could you split out the different changes into separate PRs so they can be merged independently?

**Comment by gilv** [comment]
@calvinjia I will split the commits into different PRs.

**Comment by gilv** [comment]
Will split into different PRs
